### ISSUE TITLE
fixing -Wsign-compare warning on GCC

### ISSUE THREAD

**Issue by darealshinji** [p-ranav/argparse#279]
I get a warning about sign compare:
```
argparse.hpp: In function ‘std::ostream& argparse::operator<<(std::ostream&, const Argument&)’:
argparse.hpp:686:53: warning: comparison of integer expressions of different signedness: ‘int’ and ‘const std::__cxx11::basic_string<char>::size_type’ {aka ‘const long unsigned int’} [-Wsign-compare]
  686 |     while ((pos = argument.m_help.find('\n', prev)) != std::string::npos) {
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
```

This is because at line 680 `auto pos` is recognized as signed int instead of an unsigned type.
I recommend marking the values of `pos` and `prev` as unsigned or setting their type explicitly to `size_t`:
``` C++
    auto pos = 0u;
    auto prev = 0u;
```
``` C++
    size_t pos = 0;
    size_t prev = 0;
```

See also the comments in this PR: https://github.com/p-ranav/argparse/pull/276#discussion_r1277172415

**Comment by darealshinji** [comment]
Forget that fix. After that `-h` and `--help` and automatic help messages don't work anymore. This might explain https://github.com/p-ranav/argparse/issues/242

**Comment by p-ranav** [comment]
Fixed by #292
